example/src/main/java/com/dsc/formbuilder/screens/survey/SurveyViewmodel.kt

```diff
@@ -4,21 +4,93 @@ import androidx.compose.runtime.MutableState
 import androidx.compose.runtime.State
 import androidx.compose.runtime.mutableStateOf
 import androidx.lifecycle.ViewModel
+import com.dsc.form_builder.*
 
-class SurveyViewmodel: ViewModel(){
+class SurveyViewmodel : ViewModel() {
 
     private val _screen: MutableState<Int> = mutableStateOf(0)
     val screen: State<Int> = _screen
 
-    fun navigate(screen: Int){
+    fun navigate(screen: Int) {
         _screen.value = screen
     }
 
     fun validateSurvey() {
         // TODO: Implement validation
     }
 
-    fun validateScreen(screen: Int){
+    fun validateScreen(screen: Int) {
         // TODO: Implement validation
     }
+
```

Comment:
I see what you were talking about but we only need one formstate. It should hold all the fields available in the form so you basically need to merge all the three variables you created. It can hold fields of multiple types.

Reply:
Noted